nushell_completions.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2013-2022 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cobra
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func (c *Command) GenNushellCompletion(w io.Writer, includeDesc bool) error {
+	buf := new(bytes.Buffer)
+	WriteStringAndCheck(buf, fmt.Sprintf(`
+# An external completer that works with any cobra based
+# command line application (e.g. kubectl, minikube)
+let cobra_completer = {|spans|
+	let ShellCompDirectiveError = %[1]d
+	let ShellCompDirectiveNoSpace = %[2]d
+	let ShellCompDirectiveNoFileComp = %[3]d
+	let ShellCompDirectiveFilterFileExt = %[4]d
+	let ShellCompDirectiveFilterDirs = %[5]d
+
+	let cmd = $spans | first 
+	let rest = $spans | skip
+
+	def exec_complete [
+		spans: list<string>
+	] {
+		# This will catch the stderr message related to the directive and any other errors,
+		# such as the command not being a cobra based command
+		let result = do --ignore-errors { cobra_active_help=0 run-external $cmd "__complete" ...$spans | complete }
```

Comment:
`cobra_active_help` -> `COBRA_ACTIVE_HELP`.
It doesn't work if it is not in uppercase.

Reply:
hmm.. I am not sure how that go lowercases 🤷🏻